gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java

```diff
@@ -76,6 +81,13 @@ private static ResponseMap toResponseMap(@Nullable Map<String, Object> docRespon
       } else if (obj instanceof List) {
         MultipleChoiceResponse.fromList((List<String>) obj)
             .ifPresent(r -> responses.putResponse(fieldId, r));
+      } else if (obj instanceof Map) {
+        HashMap<String, Long> map = (HashMap<String, Long>) obj;
```

Comment:
Can we use Firebase `Timestamp` type for dates and `String` for times here instead of `Map`? In that case you may need to pass in the Form object to be able to determine whether the value is a string or a time.  Wdyt? Would prefer this so that the db is future-proof; if we do this now we'll need to maintain the Map representation for backward compatibility forever.